### PR TITLE
Only sign reviewed files (bug 1156221)

### PR DESCRIPTION
**Pull request by magopian**
Fixes [bug 1156221](https://bugzilla.mozilla.org/show_bug.cgi?id=1156221)

Took the opportunity to do some cleanup/refactor.

### CODE REVIEW

File: apps/editors/helpers.py [hunk 1]
```diff
@@ -750,12 +750,12 @@ def process_public(self):
...
         status = self.addon.status
 
-        # Sign addon.
-        self.version.sign_files()
-
         self.set_files(amo.STATUS_PUBLIC, self.data['addon_files'],
                        copy_to_mirror=True)
 
+        # Sign addon.
+        self.version.sign_files()
```
Comment (robhudson) [inline apps/editors/helpers.py]: Why this change? Is there a chance someone could get the file before it is signed?

Comment (magopian) [inline apps/editors/helpers.py]: We only sign files that are prelim or fully reviewed. I've put the other calls to `sign_files` in the correct order in the other places in this file, but not that one. I'm now fixing this mistake ;)

I'll need to change that though for https://bugzilla.mozilla.org/show_bug.cgi?id=1156333, but in another PR

File: lib/crypto/packaged.py [hunk 2]
```diff
@@ -32,95 +31,71 @@ def get_endpoint(file_obj):
...
+    # We only sign files that have been reviewed.
+    if file_obj.status not in [amo.STATUS_LITE, amo.STATUS_PUBLIC]:
+        log.info("Not signing file {0}: it isn't reviewed".format(file_obj.pk))
+        return
+
+    endpoint = get_endpoint(file_obj)
+    if not endpoint:  # Signing not enabled.
+        log.info('Not signing file {0}: no active endpoint'.format(
+            file_obj.pk))
+        return
```
Comment (robhudson) [inline lib/crypto/packaged.py]: It seems like this should go first to me. If it isn't configured none of the others matter.

Comment (magopian) [inline lib/crypto/packaged.py]: you're right ;)

File: lib/crypto/tasks.py [hunk 3]
```diff
@@ -41,15 +38,27 @@ def sign_addons(addon_ids, force=False, **kw):
...
-            # Now update the Version model.
+                signed = signed or bool(sign_file(file_obj))
+            except:
+                log.error('Failed signing version {0}'.format(version.pk),
+                          exc_info=True)
+                # Revert the version bump, restore the backup.
+                shutil.copy(backup_path, file_obj.file_path)
+        # Now update the Version model, if we signed at least one file.
+        if signed:
             version.update(version='{0}.1'.format(version.version))
```
Comment (robhudson) [inline lib/crypto/tasks.py]: I'm assuming the other PR will update this to use the newer `.1-signed` name?

Comment (magopian) [inline lib/crypto/tasks.py]: Yes, #517 is in charge of doing that change.
